src/storage/storage_service.ml

```diff
@@ -22,9 +25,10 @@ module Make (Log : Log.Sig.SERVICE) (Repo : REPO) : SERVICE = struct
   let delete ctx ~id =
     let* file = find ctx ~id in
     let blob_id = StoredFile.blob file in
-    let* () = Repo.delete_blob ctx ~id:blob_id in
-    let* () = Repo.delete_file ctx ~id:file.file.id in
-    Lwt.return ()
+    DbService.atomic ctx (fun ctx ->
+        let* () = Repo.delete_file ctx ~id:file.file.id in
+        let* () = Repo.delete_blob ctx ~id:blob_id in
```

Comment:
@aronerben You could just return the value of the last call instead of doing `Lwt.return`

Reply:
Cheers